src/tests/system/tests/readme.rst

```diff
@@ -0,0 +1,115 @@
+* test_passkey.py, contains authentication tests, using virtual passkeys.
+* test_smartcards.py, contains authentication tests, using virtual smart cards.
+* test_identity.py, contains tests resolving users, groups, group memberships and the object attributes.
+* test_lookup.py, similar to identity tests but the tests focus on the client configuration.
+* test_netgroup.py, contains all tests related to netgroups.
+* test_access_control.py, contains simple, LDAP or AD filter access control tests.
+* test_gpo.py, contains access control tests managed by group policy objects (gpo) in AD.
+* test_cache.py, contains LDB cache tests.
+* test_memcache.py, contains in-memory cache tests.
+* test_files.py, legacy files provider tests.
+* test_proxy.py, proxy provider tests.
+* test_failover.py, any provider resolution and connectivity tests.
+* test_ldap.py, any test that is specific to LDAP.
+* test_ad.py, any test that is specific to AD.
+* test_ipa.py, any test that is specific to IPA.
+* test_ipa_trust.py, any IPA trust test
+* test_krb.py, any test that is specific to kerberos.
+* test_service.py, any test that is checking the SSSD service, process and child processes.
+* test_kcm.py, contains kerberos cache manager tests.
+* test_autofs.py, contains auto file system (autofs) or automount tests.
+* test_sudo.py, contains sudo privilege escalation tests.
+* test_tools.py, contains any cli tool tests, like sss_obfuscate.
+* test_sss_override.py, contains sss_override CLI tests.
+* test_sssctl.py, contains sssctl CLI tests.
+* test_infopipe.py, contains any d-bus infopipe tests.
+* test_python_modules.py, contains integration tests for sss python modules.
+* test_services.py, contains tests specific to the SSSD and processes.
+* test_responders.py, contains responders tests, NSS, PAM and others.
+
+.. note::
+
+    Samba Active Directory is the only AD topology used when testing upstream. Please use 'AnyAD' topology if early testing is needed.
+
+Test Scope
+**********
+
+Tests are written to check a specific user scenario, parameter, issue or bug. Often it will be faster to merge tests into one, or when test cases overlap. For simplicity, every test case is small and covers a specific case.
+
+Test Naming
+***********
+
+Pytest requires that tests names and file names are pre-fixed with *test_*. Test names have been standardize to then contain the file name and describe what the test code does and may not match related bugs, issues or tickets.
+
+Docstrings
+**********
+
+Each test is required to contain the following docstring fields.
+
+* title: Required, descriptive abbreviated test name, the name should give you a good idea of what it covers.
+* description: Optional, provide more detail about the test if necessary.
+* setup: Required, incremented list of steps needed before the test is executed.
+* steps: Required, incremented list of the test steps. The first step should be the the test.
```

Comment:
`the the`

What do you mean by `The first step should be the test`?

Reply:
That is unclear, if the test is looking up a user, the step should begin there. Where as if the test is testing the startup of SSSD, the test should begin with SSSD starting.  Reworded, 

`
* setup: Required, incremented list of steps needed before the test scenario can be executed.
* steps: Required, incremented list of the test steps, beginning with the test scenario.
 `
?

Reply:
Good enough for me.